server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -83,15 +85,22 @@ public MappedFieldType(String name, boolean isIndexed, boolean hasDocValues, Tex
      * Return a fielddata builder for this field
      *
      * @param fullyQualifiedIndexName the name of the index this field-data is build for
-     *
+     * @param searchLookup a {@link SearchLookup} supplier to allow for accessing other fields values in the context of runtime fields
      * @throws IllegalArgumentException if the fielddata is not supported on this type.
      * An IllegalArgumentException is needed in order to return an http error 400
      * when this error occurs in a request. see: {@link org.elasticsearch.ExceptionsHelper#status}
      */
-    public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
+    public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
         throw new IllegalArgumentException("Fielddata is not supported on field [" + name() + "] of type [" + typeName() + "]");
     }
 
+    /**
+     * Return true for field types that are runtime fields implementation, false otherwise
+     */
+    public boolean isRuntimeField() {
```

Comment:
I wonder if this is the right way to do since we may forgot to override this function where needed ?
Could we instead rely on some inheritance, runtime field type must extend a specific class to be considered runtime ?

Reply:
The idea is that runtime fields will have a common base class that overrides this method. This method is basically here to prevent having to perform instanceof checks to figure out whether something is a runtime field.

Reply:
ok fine with me if we have a long-term solution in the main PR